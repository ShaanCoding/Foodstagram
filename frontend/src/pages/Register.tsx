import styles from '../styles/Register.module.css'
import Foostaram from '../images/Foostaram.svg'
import InputField from '../components/form/InputField'
import SubmitButton from '../components/form/SubmitButton'
import UseRegisterMutation from '../api/UseRegisterMutation'
import Form from '../components/form/Form'
import { AxiosError } from 'axios'

const Register = () => {
	const registerMutation = UseRegisterMutation()

	if (registerMutation.isError) {
		console.log(registerMutation.error)
	}

	return (
		<div className="m-12 w-4/5 h-full ml-auto mr-auto flex max-w-[750px]">
			<div className="hidden md:block flex-auto w-32 mr-8 mt-6">
				<img
					alt="Promotional screenshots of the website"
					src="/images/login/login-screenshot.png"
					width={380}
					height={581}
				/>
			</div>
			<div className={`flex-auto w-14`}>
				<div className={`py-12 px-8 bg-white border ${styles.greyBorder}`}>
					<img
						alt="Foodstaram Logo"
						className="h-10 ml-auto mr-auto"
						src={Foostaram}
					/>
					<p className="font-medium text-lg text-gray-500 text-center mt-4 mb-8">
						Sign up to see food photos from your friends.
					</p>
					{registerMutation.isError && (
						<div className="my-8 bg-red-300 rounded-lg p-4 text-center">
							{`Failed to create account. ${
								registerMutation.error instanceof AxiosError &&
								registerMutation.error?.response?.status === 400
									? registerMutation.error.response.data.message
									: ''
							}`}
						</div>
					)}

					{registerMutation.isSuccess && (
						<div className="my-8 bg-green-300 rounded-lg p-4 text-center">
							{`Welcome, you've succesfully created an account. Please check your email for a verification code!`}
						</div>
					)}
					<Form
						onSubmit={(data) => {
							registerMutation.mutate({
								fullName: data['fullName'],
								username: data['username'],
								email: data['email'],
								password: data['password'],
							})
						}}
					>
						<InputField
							placeholder="Email address"
							type="email"
							name="email"
							autoComplete="foostagram-email"
							required
						/>
						<InputField
							placeholder="Full Name"
							name="fullName"
							autoComplete="false"
							required
						/>
						<InputField
							placeholder="Username"
							name="username"
							autoComplete="false"
							required
						/>
						<InputField
							placeholder="Password"
							type="password"
							name="password"
							autoComplete="foostagram-password"
							required
						/>
						<p className="text-xs text-center text-gray-400 mt-2 mb-6">
							By signing up, you agree to our{' '}
							<a className="font-medium text-gray-500" href="/">
								Terms
							</a>
							,{' '}
							<a className="font-medium text-gray-500" href="/">
								Privacy Policy
							</a>{' '}
							and{' '}
							<a className="font-medium text-gray-500" href="/">
								Cookies Policy
							</a>
							.
						</p>
						<SubmitButton text="Sign Up" loading={registerMutation.isLoading} />
					</Form>
				</div>
				<div className={`mt-2 py-4 px-8 bg-white border ${styles.greyBorder}`}>
					<p className="text-sm text-center">
						Have an account?{' '}
						<a className="text-insta-green" href="/login">
							Log In
						</a>
					</p>
				</div>
			</div>
		</div>
	)
}

export default Register
