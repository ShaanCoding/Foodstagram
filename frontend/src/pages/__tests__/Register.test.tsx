import { QueryClient, QueryClientProvider } from 'react-query'
import { BrowserRouter, Router } from 'react-router-dom'
import renderer from 'react-test-renderer'
import Register from '../Register'

const queryClient = new QueryClient()

it('Register page renders correctly', () => {
	const tree = renderer
		.create(
			<QueryClientProvider client={queryClient}>
				<BrowserRouter>
					<Register />
				</BrowserRouter>
			</QueryClientProvider>
		)
		.toJSON()
	expect(tree).toMatchSnapshot()
})
