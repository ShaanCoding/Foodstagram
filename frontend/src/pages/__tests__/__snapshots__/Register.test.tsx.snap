// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register page renders correctly 1`] = `
<div
  className="m-12 w-4/5 h-full ml-auto mr-auto flex max-w-[750px]"
>
  <div
    className="hidden md:block flex-auto w-32 mr-8 mt-6"
  >
    <img
      alt="Promotional screenshots of the website"
      height={581}
      src="/images/login/login-screenshot.png"
      width={380}
    />
  </div>
  <div
    className="flex-auto w-14"
  >
    <div
      className="py-12 px-8 bg-white border undefined"
    >
      <img
        alt="Foodstaram Logo"
        className="h-10 ml-auto mr-auto"
        src="test-file-stub"
      />
      <p
        className="font-medium text-lg text-gray-500 text-center mt-4 mb-8"
      >
        Sign up to see food photos from your friends.
      </p>
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="foostagram-email"
          className="p-2 w-full rounded-sm mb-2 border hover:border-gray-500 focus-visible:border text-xs bg-white undefined"
          initialValue=""
          name="email"
          onChange={[Function]}
          placeholder="Email address"
          required={true}
          type="email"
          value=""
        />
        <input
          autoComplete="false"
          className="p-2 w-full rounded-sm mb-2 border hover:border-gray-500 focus-visible:border text-xs bg-white undefined"
          initialValue=""
          name="fullName"
          onChange={[Function]}
          placeholder="Full Name"
          required={true}
          value=""
        />
        <input
          autoComplete="false"
          className="p-2 w-full rounded-sm mb-2 border hover:border-gray-500 focus-visible:border text-xs bg-white undefined"
          initialValue=""
          name="username"
          onChange={[Function]}
          placeholder="Username"
          required={true}
          value=""
        />
        <input
          autoComplete="foostagram-password"
          className="p-2 w-full rounded-sm mb-2 border hover:border-gray-500 focus-visible:border text-xs bg-white undefined"
          initialValue=""
          name="password"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
          value=""
        />
        <p
          className="text-xs text-center text-gray-400 mt-2 mb-6"
        >
          By signing up, you agree to our
           
          <a
            className="font-medium text-gray-500"
            href="/"
          >
            Terms
          </a>
          ,
           
          <a
            className="font-medium text-gray-500"
            href="/"
          >
            Privacy Policy
          </a>
           
          and
           
          <a
            className="font-medium text-gray-500"
            href="/"
          >
            Cookies Policy
          </a>
          .
        </p>
        <button
          className="w-full bg-insta-green text-white text-sm p-[5px] rounded-md font-medium"
          disabled={false}
          loading={false}
          text="Sign Up"
          type="submit"
        >
          Sign Up
        </button>
      </form>
    </div>
    <div
      className="mt-2 py-4 px-8 bg-white border undefined"
    >
      <p
        className="text-sm text-center"
      >
        Have an account?
         
        <a
          className="text-insta-green"
          href="/login"
        >
          Log In
        </a>
      </p>
    </div>
  </div>
</div>
`;
